ui/src/reusable_ui/components/card_select/CardSelectCard.tsx

```diff
@@ -1,86 +1,59 @@
+  id: string
+  name?: string
+  label: string
+  image?: string
+  checked?: boolean
+  disabled?: boolean
+  onClick: () => void
 }
 
 @ErrorHandling
-class CardSelectCard extends PureComponent<CardSelectCardProps, State> {
-  public static defaultProps: Partial<CardSelectCardProps> = {
-    checked: false,
-    disabled: false,
-  }
-
-  constructor(props) {
-    super(props)
-    this.state = {
-      checked: this.props.checked,
-    }
-  }
-
+class CardSelectCard extends PureComponent<Props> {
   public render() {
-    const {id, label, disabled} = this.props
-    const {checked} = this.state
+    const {id, label, checked, name, disabled} = this.props
 
     return (
-      <div className="card-select--card-holder">
-        <div
-          data-toggle="card_toggle"
-          onClick={this.toggleChecked}
-          className={classnames('card-select--card', {
-            'card-select--checked': checked,
-            'card-select--disabled': disabled,
-            'card-select--active': !disabled,
-          })}
-        >
-          <label className="card-select--container">
-            <input
-              id={`card_select_${id}`}
-              name={`card_select_${id}`}
-              type="checkbox"
-              value={id}
-              checked={checked}
-              disabled={disabled}
-            />
-            <span
-              className={classnames(
-                'card-select--checkmark',
-                'icon',
-                'checkmark',
-                {
-                  'card-select--checked': checked,
-                }
-              )}
-            />
-            <div className="card-select--image">{this.cardImage}</div>
-            <span className="card-select--label">{label}</span>
-          </label>
-        </div>
+      <div
+        data-toggle="card_toggle"
+        onClick={!disabled && this.handleClick}
+        className={classnames('card-select--card', {
+          'card-select--checked': checked,
+          'card-select--disabled': disabled,
+          'card-select--active': !disabled,
+        })}
+      >
+        <label className="card-select--container">
+          <input
+            id={`card_select_${id}`}
+            name={name}
+            type="checkbox"
+            value={id}
+            checked={checked || false}
+            disabled={disabled || false}
+          />
+          <span
+            className={classnames(
+              'card-select--checkmark',
+              'icon',
+              'checkmark',
+              {
+                'card-select--checked': checked,
+              }
+            )}
+          />
+          <div className="card-select--image">{this.cardImage}</div>
+          <span className="card-select--label">{label}</span>
+        </label>
       </div>
     )
   }
 
-  private toggleChecked = e => {
-    const {disabled} = this.props
-
-    if (disabled) {
-      return
-    }
-
-    const {checked} = this.state
-    e.preventDefault()
```

Comment:
shouldn't we be preventingDefault even if disabled? as it is if disabled we will not reach this line.

Reply:
Updated